src/pages/main.tsx

```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import Header, { links } from '../components/header';
 import SearchBar from '../components/searchBar';
 import ItemsCardsList from '../components/itemsCardsList';
```

Comment:
You have a mix of paths, better to have a rule to control it, like you can use relative paths for the same directory and use absolute paths for any other files